### PR TITLE
Consume first TFM for `TagHelperDescriptor` resolution.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- This makes RazorTooling consistent with the .cshtml editor.
- Added `project.json` tests to validate new behavior of throwing an appropriate error/selecting the first TFM.

#49

### CODE REVIEW

File: src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs [hunk 1]
```diff
@@ -82,6 +74,23 @@ internal static void Register(CommandLineApplication app)
             });
         }
 
+        /// <summary>
+        /// Internal for testing.
+        /// </summary>
+        public static ProjectContext ResolveProjectContext(string projectFilePath)
```
Comment (pranavkm) [inline src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs]: `internal`?

File: src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs [hunk 2]
```diff
@@ -82,6 +74,23 @@ internal static void Register(CommandLineApplication app)
             });
         }
 
+        /// <summary>
+        /// Internal for testing.
```
Comment (pranavkm) [inline src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs]: > // Internal for unit testing.

File: src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs]: Would rather not have an `InternalsVisibleTo` and no one else consumes this :smile:

Comment (NTaylorMullen) [inline src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs]: Can change comment to `Only for testing`.

File: test/dotnet-razor-tooling.Test/ResolveTagHelpersCommandTest.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        public void ResolveProjectContext_ResolvesProjectContextsCorrectly()
+        {
+            // Arrange
+            var projectFilePath = "TestFiles/dnxcoreproject.json";
+
+            // Act
+            var projectContext = ResolveTagHelpersCommand.ResolveProjectContext(projectFilePath);
+
+            // Assert
+            Assert.Equal("DNXCore", projectContext.TargetFramework.Framework, StringComparer.Ordinal);
```
Comment (pranavkm) [inline test/dotnet-razor-tooling.Test/ResolveTagHelpersCommandTest.cs]: I always thought `Assert.Equal` used Ordinal comparison by default. Is that not the case?

File: src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs (hunk 1 above)
Comment (pranavkm) [inline src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs]: Ok. Maybe remove the comment in that case entirely in that case. No harm in it being public

**Comment by pranavkm** [comment]
:shipit:

File: src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/dotnet-razor-tooling/Internal/ResolveTagHelpersCommand.cs]: kk

File: test/dotnet-razor-tooling.Test/ResolveTagHelpersCommandTest.cs (hunk 3 above)
Comment (NTaylorMullen) [inline test/dotnet-razor-tooling.Test/ResolveTagHelpersCommandTest.cs]: It does, will remove the extra.

**Pull request merged**